### ISSUE TITLE
Option to copy VM files without combining them into a Zip/Rar file

### ISSUE THREAD

**Issue by GMcDonnell** [ColiseoSoftware/hypervbackup#2]
Hello and thanks for your improvements!
It would be really useful to have an option to copy the VMs without adding them to a Zip file - basically creating a duplicate folder structure at the backup target.

**Comment by pablopioli** [comment]
I would like to clearly define the requirements. My view is like this: in the output folder it will create a new folder using the virtual machine name, and inside it a new folder for every backup using the current time. Something like that.

\share\backups\myvmname\20160115-183025

I would like to hear comments.

**Comment by GMcDonnell** [comment]
I like the idea of each VM being in a separate folder with the same name as that VM. It might also make sense to follow the current structure that's used for compressed files as well. Use "--output" to define the base path as is done now and then use "--outputformat" to specify the subfolder name. As an example consider the following command line:

```
HVBackup.exe --backup --list VM01,VM02 (--copy as files switch) --output X:\ImageBackups --outputformat "Current" (other command line options)
```

would result in a folder structure of:
X:\ImageBackups\Current\VM01\VHDs
X:\ImageBackups\Current\VM01\VMs
X:\ImageBackups\Current\VM02\VHDs
X:\ImageBackups\Current\VM02\VMs

Main reason for this is that being able to define a folder structure makes it easier to script things like moves and copies of the data.

Don't know if I'm making sense here.

**Comment by pablopioli** [comment]
There are two situations, if in the output switch you use a variable name (eg: time based) everything is OK. If you use a static location (as current in your example) all existing files in the output location will be deleted. Just to not mess things up.

On the other part, in the final folder (output\current in your example) the original layout of the virtual machines and hard drives will be used. I don't see the point in moving things around, I you need a specific layout you must use it in the source locations or you can move them later using scripts.

**Comment by GMcDonnell** [comment]
I appreciate your quick response!

The ability to specify a static location is indeed something that I need. That makes it easier to have another process copy or move the folder elsewhere.

Regarding the second item - keeping the original layout/folder structure of the virtual machine files in the target folder makes sense – that’s actually what I want but I didn’t express myself very well.

**Comment by pablopioli** [comment]
ok, this feature is planned for a 2016 March release (as I do this is my free time), but I will surely post some betas to try it.

**Comment by pablopioli** [comment]
Sorry for the delay, a lot of work ...

You can try it with this version
https://onedrive.live.com/redir?resid=6DBCDBD8B6418012!869730&authkey=!ANwcLFgdOKQXsUw&ithint=file%2c7z

use the new d (directcopy) to enable it. Hope it works as you wanted, let me know.

**Comment by GMcDonnell** [comment]
Thanks so much! I’ll test it out later this week (dealing with server patching today).

Take care –

Gary McDonnell
214-324-1230

**Comment by pablopioli** [comment]
One more thing, currently it doesnt display the progress bar.
It thought it would slow it down (personally dont think reading chunks of a file in memory with c# to copy is very performant when you back a lot of GB)

**Comment by pablopioli** [comment]
Any news on this?
If this works as expected a will release an official version.

**Comment by GMcDonnell** [comment]
I'm just looking at it today, sorry for the delay. 

I'm in the process of updating my test Hyper-V server with .Net 4.6 right now as I recall that's a requirement. Other than .Net 4.6 and the 64 bit 7z.dll do I need anything besides the two files you provided?

**Comment by pablopioli** [comment]
No, just use the last released version and replace the updates files

There was a bug in the beta listed above, use this files instead 
https://onedrive.live.com/redir?resid=6DBCDBD8B6418012!871498&authkey=!AEQ6DUz8bXhyZJU&ithint=file%2c7z

**Comment by GMcDonnell** [comment]
Sorry to be a pain but I can't seem to find the right versions of the support files. There's an error regarding AlphaVSS.Common - it's looking for version 1.2.4000.3 but I can't find that version (I found the source code for that DLL but I have no access to a compiler). Would you be able to point me to what I need?

EDIT: Never mind, I believe I have found all the files I need now.

**Comment by GMcDonnell** [comment]
Backups appear to be working fine, including the uncompressed copy (-d option).

One small item: In the "Note:" line the word "switchs" should be spelled "switches" (two instances).

**Comment by pablopioli** [comment]
Thanks for your feedback, in a few days I will publish a official released.
Corrected the misspelling too.

**Comment by pablopioli** [comment]
Released version 2.2 with this feature included, if you were using it in production there is no need to update the program, as it is exactly the same code with another version number.
